flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -3,27 +3,257 @@ import Sentry
+                result(FlutterError(code: "4", message: "A valid Dsn must be provided", details: nil) )
+            }
+
+            if let isDebug = arguments["debug"] as? Bool {
+                options.debug = isDebug
+            }
+
+            if let environment = arguments["environment"] as? String{
+                options.environment = environment
+            }
+
+            if let releaseName = arguments["release"] as? String{
+                options.releaseName = releaseName
+            }
+
+            if let enableAutoSessionTracking = arguments["enableAutoSessionTracking"] as? Bool {
+                options.enableAutoSessionTracking = enableAutoSessionTracking
+            }
+
+            if let attachStacktrace = arguments["attachStacktrace"] as? Bool {
+                options.attachStacktrace = attachStacktrace
+            }
+
+            if let diagnosticLevel = arguments["diagnosticLevel"] as? String, options.debug == true {
+
+                switch (diagnosticLevel) {
+                case "fatal", "error":
+                    options.logLevel = .error
+                    break;
+                case "debug":
+                    options.logLevel = .debug
+                    break;
+                case "warning", "info":
+                    options.logLevel = .verbose
+                    break;
+                default:
+                    options.logLevel = .none
+                    break;
+                }
+            }
+
+            if let sessionTrackingIntervalMillis = arguments["sessionTrackingIntervalMillis"] as? UInt{
+                options.sessionTrackingIntervalMillis = sessionTrackingIntervalMillis
+            }
+
+            if let dist = arguments["dist"] as? String{
+                options.dist = dist
+            }
+
+
+            if let enableAutoNativeBreadcrumbs = arguments["enableAutoNativeBreadcrumbs"] as? Bool,
+               enableAutoNativeBreadcrumbs == false {
+                options.integrations = options.integrations?.filter { (name) -> Bool in
+                    return name != "SentryAutoBreadcrumbTrackingIntegration"
+                }
+            }
+
+            if let maxBreadcrumbs = arguments["maxBreadcrumbs"] as? UInt{
+                options.maxBreadcrumbs = maxBreadcrumbs
+            }
+
+            print("------ options.integrations =======> \(String(describing: options.integrations))")
+            //print("------ packages =======> \(String(describing: arguments["packages"] as? [String]))")
+            //print("------ integrations =======> \(String(describing: arguments["integrations"] as? [String]))")
+
+            self.sentryOptions = options
+            /*
+             TODO : beforeSend alternative =>
+              - eventOrigin
+             */
+
+            /* for now, in sentry-cocoa, beforeSend is not called before captureEnvelope
+            options.beforeSend = { event in
+                self.setEventOriginTag(event: event)
```

Comment:
now this is commented but still would require to add the tag for iOS events which run the beforeSend callback.

if an event is captured on Dart/Flutter, we'd need to add before capturing the envelope.
if an event is captured on iOS, we could add just here as before

Reply:
✅